homeassistant/components/weather/__init__.py

```diff
@@ -1075,3 +1098,25 @@ async def async_update_listeners(
             converted_forecast_list = self._convert_forecast(native_forecast_list)
             for listener in self._forecast_listeners[forecast_type]:
                 listener(converted_forecast_list)
+
+
+async def async_get_forecast_service(
+    weather: WeatherEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """Get weather forecast."""
+    forecast_type = service_call.data["type"]
+    if forecast_type == "daily":
+        native_forecast_list = await weather.async_forecast_daily()
+    elif forecast_type == "hourly":
+        native_forecast_list = await weather.async_forecast_hourly()
+    else:
+        native_forecast_list = await weather.async_forecast_twice_daily()
+    if native_forecast_list is None:
+        converted_forecast_list = None
+    else:
+        # pylint: disable-next=protected-access
+        converted_forecast_list = weather._convert_forecast(native_forecast_list)
+    return {
+        "type": forecast_type,
+        "forecast": cast(JsonValueType, converted_forecast_list),
```

Comment:
I can't get the typing to work. Mypy complains also after updating `WeatherEntity._convert_forecast` to return `list[JsonValueType]`.

Reply:
Above works for me and seems also for CI so maybe something locally for you that's off?